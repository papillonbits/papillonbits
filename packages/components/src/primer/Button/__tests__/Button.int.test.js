import {
  defaultNormal,
  defaultSmall,
  defaultLarge,
  defaultInactive,
  defaultIconLeftAligned,
  defaultIconRightAligned,
  defaultLink,
  primary,
  primarySmall,
  primaryLarge,
  primaryInactive,
  primaryIconLeftAligned,
  primaryIconRightAligned,
  danger,
  dangerSmall,
  dangerLarge,
  dangerInactive,
  dangerIconLeftAligned,
  dangerIconRightAligned,
  outline,
  outlineSmall,
  outlineLarge,
  outlineInactive,
  outlineIconLeftAligned,
  outlineIconRightAligned,
  blue,
  blueSmall,
  blueLarge,
  blueInactive,
  blueIconLeftAligned,
  blueIconRightAligned,
  orange,
  orangeSmall,
  orangeLarge,
  orangeInactive,
  orangeIconLeftAligned,
  orangeIconRightAligned,
} from './Button.int.story'

describe('<Button />', () => {
  describe('Render', () => {
    test('must match defaultNormal()', () => {
      expect(global.renderToJSON(defaultNormal())).toMatchSnapshot()
    })

    test('must match defaultSmall()', () => {
      expect(global.renderToJSON(defaultSmall())).toMatchSnapshot()
    })

    test('must match defaultLarge()', () => {
      expect(global.renderToJSON(defaultLarge())).toMatchSnapshot()
    })

    test('must match defaultInactive()', () => {
      expect(global.renderToJSON(defaultInactive())).toMatchSnapshot()
    })

    test('must match defaultIconLeftAligned()', () => {
      expect(global.renderToJSON(defaultIconLeftAligned())).toMatchSnapshot()
    })

    test('must match defaultIconRightAligned()', () => {
      expect(global.renderToJSON(defaultIconRightAligned())).toMatchSnapshot()
    })

    test('must match defaultLink()', () => {
      expect(global.renderToJSON(defaultLink())).toMatchSnapshot()
    })

    test('must match primary()', () => {
      expect(global.renderToJSON(primary())).toMatchSnapshot()
    })

    test('must match primarySmall()', () => {
      expect(global.renderToJSON(primarySmall())).toMatchSnapshot()
    })

    test('must match primaryLarge()', () => {
      expect(global.renderToJSON(primaryLarge())).toMatchSnapshot()
    })

    test('must match primaryInactive()', () => {
      expect(global.renderToJSON(primaryInactive())).toMatchSnapshot()
    })

    test('must match primaryIconLeftAligned()', () => {
      expect(global.renderToJSON(primaryIconLeftAligned())).toMatchSnapshot()
    })

    test('must match primaryIconRightAligned()', () => {
      expect(global.renderToJSON(primaryIconRightAligned())).toMatchSnapshot()
    })

    test('must match danger()', () => {
      expect(global.renderToJSON(danger())).toMatchSnapshot()
    })

    test('must match dangerSmall()', () => {
      expect(global.renderToJSON(dangerSmall())).toMatchSnapshot()
    })

    test('must match dangerLarge()', () => {
      expect(global.renderToJSON(dangerLarge())).toMatchSnapshot()
    })

    test('must match dangerInactive()', () => {
      expect(global.renderToJSON(dangerInactive())).toMatchSnapshot()
    })

    test('must match dangerIconLeftAligned()', () => {
      expect(global.renderToJSON(dangerIconLeftAligned())).toMatchSnapshot()
    })

    test('must match dangerIconRightAligned()', () => {
      expect(global.renderToJSON(dangerIconRightAligned())).toMatchSnapshot()
    })

    test('must match outline()', () => {
      expect(global.renderToJSON(outline())).toMatchSnapshot()
    })

    test('must match outlineSmall()', () => {
      expect(global.renderToJSON(outlineSmall())).toMatchSnapshot()
    })

    test('must match outlineLarge()', () => {
      expect(global.renderToJSON(outlineLarge())).toMatchSnapshot()
    })

    test('must match outlineInactive()', () => {
      expect(global.renderToJSON(outlineInactive())).toMatchSnapshot()
    })

    test('must match outlineIconLeftAligned()', () => {
      expect(global.renderToJSON(outlineIconLeftAligned())).toMatchSnapshot()
    })

    test('must match outlineIconRightAligned()', () => {
      expect(global.renderToJSON(outlineIconRightAligned())).toMatchSnapshot()
    })

    test('must match blue()', () => {
      expect(global.renderToJSON(blue())).toMatchSnapshot()
    })

    test('must match blueSmall()', () => {
      expect(global.renderToJSON(blueSmall())).toMatchSnapshot()
    })

    test('must match blueLarge()', () => {
      expect(global.renderToJSON(blueLarge())).toMatchSnapshot()
    })

    test('must match blueInactive()', () => {
      expect(global.renderToJSON(blueInactive())).toMatchSnapshot()
    })

    test('must match blueIconLeftAligned()', () => {
      expect(global.renderToJSON(blueIconLeftAligned())).toMatchSnapshot()
    })

    test('must match blueIconRightAligned()', () => {
      expect(global.renderToJSON(blueIconRightAligned())).toMatchSnapshot()
    })

    test('must match orange()', () => {
      expect(global.renderToJSON(orange())).toMatchSnapshot()
    })

    test('must match orangeSmall()', () => {
      expect(global.renderToJSON(orangeSmall())).toMatchSnapshot()
    })

    test('must match orangeLarge()', () => {
      expect(global.renderToJSON(orangeLarge())).toMatchSnapshot()
    })

    test('must match orangeInactive()', () => {
      expect(global.renderToJSON(orangeInactive())).toMatchSnapshot()
    })

    test('must match orangeIconLeftAligned()', () => {
      expect(global.renderToJSON(orangeIconLeftAligned())).toMatchSnapshot()
    })

    test('must match orangeIconRightAligned()', () => {
      expect(global.renderToJSON(orangeIconRightAligned())).toMatchSnapshot()
    })
  })
})
