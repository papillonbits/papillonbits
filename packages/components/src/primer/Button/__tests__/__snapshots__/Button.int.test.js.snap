// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render must match blue() 1`] = `
<div>
  <button
    class="btn btn-blue"
    type="button"
  >
    Blue button
  </button>
</div>
`;

exports[`<Button /> Render must match blueIconLeftAligned() 1`] = `
<div>
  <button
    class="btn btn-blue"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Blue icon button
  </button>
</div>
`;

exports[`<Button /> Render must match blueIconRightAligned() 1`] = `
<div>
  <button
    class="btn btn-blue"
    type="button"
  >
    Blue icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match blueInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn btn-blue button-inactive"
    disabled=""
    type="button"
  >
    Blue inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match blueLarge() 1`] = `
<div>
  <button
    class="btn btn-large btn-blue"
    type="button"
  >
    Blue large button
  </button>
</div>
`;

exports[`<Button /> Render must match blueSmall() 1`] = `
<div>
  <button
    class="btn btn-sm btn-blue"
    type="button"
  >
    Blue small button
  </button>
</div>
`;

exports[`<Button /> Render must match danger() 1`] = `
<div>
  <button
    class="btn btn-danger"
    type="button"
  >
    Danger button
  </button>
</div>
`;

exports[`<Button /> Render must match dangerIconLeftAligned() 1`] = `
<div>
  <button
    class="btn btn-danger"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Danger icon button
  </button>
</div>
`;

exports[`<Button /> Render must match dangerIconRightAligned() 1`] = `
<div>
  <button
    class="btn btn-danger"
    type="button"
  >
    Danger icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match dangerInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn btn-danger button-inactive"
    disabled=""
    type="button"
  >
    Danger inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match dangerLarge() 1`] = `
<div>
  <button
    class="btn btn-large btn-danger"
    type="button"
  >
    Danger large button
  </button>
</div>
`;

exports[`<Button /> Render must match dangerSmall() 1`] = `
<div>
  <button
    class="btn btn-sm btn-danger"
    type="button"
  >
    Danger small button
  </button>
</div>
`;

exports[`<Button /> Render must match defaultIconLeftAligned() 1`] = `
<div>
  <button
    class="btn"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Default icon button
  </button>
</div>
`;

exports[`<Button /> Render must match defaultIconRightAligned() 1`] = `
<div>
  <button
    class="btn"
    type="button"
  >
    Default icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match defaultInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn button-inactive"
    disabled=""
    type="button"
  >
    Default inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match defaultLarge() 1`] = `
<div>
  <button
    class="btn btn-large"
    type="button"
  >
    Default large button
  </button>
</div>
`;

exports[`<Button /> Render must match defaultLink() 1`] = `
<div>
  <a
    class="btn"
    href="#url"
    role="button"
  >
    Default link button
  </a>
</div>
`;

exports[`<Button /> Render must match defaultNormal() 1`] = `
<div>
  <button
    class="btn"
    type="button"
  >
    Default normal button
  </button>
</div>
`;

exports[`<Button /> Render must match defaultSmall() 1`] = `
<div>
  <button
    class="btn btn-sm"
    type="button"
  >
    Default small button
  </button>
</div>
`;

exports[`<Button /> Render must match orange() 1`] = `
<div>
  <button
    class="btn btn-orange"
    type="button"
  >
    Orange button
  </button>
</div>
`;

exports[`<Button /> Render must match orangeIconLeftAligned() 1`] = `
<div>
  <button
    class="btn btn-orange"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Orange icon button
  </button>
</div>
`;

exports[`<Button /> Render must match orangeIconRightAligned() 1`] = `
<div>
  <button
    class="btn btn-orange"
    type="button"
  >
    Orange icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match orangeInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn btn-orange button-inactive"
    disabled=""
    type="button"
  >
    Orange inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match orangeLarge() 1`] = `
<div>
  <button
    class="btn btn-large btn-orange"
    type="button"
  >
    Orange large button
  </button>
</div>
`;

exports[`<Button /> Render must match orangeSmall() 1`] = `
<div>
  <button
    class="btn btn-sm btn-orange"
    type="button"
  >
    Orange small button
  </button>
</div>
`;

exports[`<Button /> Render must match outline() 1`] = `
<div>
  <button
    class="btn btn-outline"
    type="button"
  >
    Outline button
  </button>
</div>
`;

exports[`<Button /> Render must match outlineIconLeftAligned() 1`] = `
<div>
  <button
    class="btn btn-outline"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Outline icon button
  </button>
</div>
`;

exports[`<Button /> Render must match outlineIconRightAligned() 1`] = `
<div>
  <button
    class="btn btn-outline"
    type="button"
  >
    Outline icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match outlineInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn btn-outline button-inactive"
    disabled=""
    type="button"
  >
    Outline inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match outlineLarge() 1`] = `
<div>
  <button
    class="btn btn-large btn-outline"
    type="button"
  >
    Outline large button
  </button>
</div>
`;

exports[`<Button /> Render must match outlineSmall() 1`] = `
<div>
  <button
    class="btn btn-sm btn-outline"
    type="button"
  >
    Outline small button
  </button>
</div>
`;

exports[`<Button /> Render must match primary() 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Primary button
  </button>
</div>
`;

exports[`<Button /> Render must match primaryIconLeftAligned() 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    <mock-file
      aria-hidden="true"
      classname="mr-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
    Primary icon button
  </button>
</div>
`;

exports[`<Button /> Render must match primaryIconRightAligned() 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Primary icon button
    <mock-file
      aria-hidden="true"
      classname="ml-2"
      height="16"
      role="img"
      style="display: inline-block; fill: currentColor; user-select: none; vertical-align: text-bottom;"
      viewbox="0 0 16 16"
      width="16"
    />
  </button>
</div>
`;

exports[`<Button /> Render must match primaryInactive() 1`] = `
<div>
  <button
    aria-disabled="true"
    class="btn btn-primary button-inactive"
    disabled=""
    type="button"
  >
    Primary inactive button
  </button>
</div>
`;

exports[`<Button /> Render must match primaryLarge() 1`] = `
<div>
  <button
    class="btn btn-large btn-primary"
    type="button"
  >
    Primary large button
  </button>
</div>
`;

exports[`<Button /> Render must match primarySmall() 1`] = `
<div>
  <button
    class="btn btn-sm btn-primary"
    type="button"
  >
    Primary small button
  </button>
</div>
`;
